src/Mutation/PostObjectCreate.php

```diff
@@ -208,11 +209,16 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
                 throw new UserError( sprintf( __( 'Sorry, you are not allowed to create %1$s', 'wp-graphql' ), $post_type_object->graphql_plural_name ) );
             }
 
-            /**
+           /**
              * If the post being created is being assigned to another user that's not the current user, make sure
              * the current user has permission to edit others posts for this post_type
              */
-            if ( ! empty( $input['authorId'] ) && get_current_user_id() !== $input['authorId'] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+            $authorID = absint($input['authorId']);
+            if ($id_components = Relay::fromGlobalId ( $input['authorId'] ) && is_array( $id_components ) && ! empty( $id_components['id'])) {
+	            $authorID = $id_components['id'];
+	        }
+
+            if ( ! empty( $authorID ) && get_current_user_id() !== $authorID && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
```

Comment:
Please break up each of these checks on their own line to make this more readable. Like:
```suggestion
            if ( ! empty( $authorID ) && 
                get_current_user_id() !== $authorID && 
                ! current_user_can( $post_type_object->cap->edit_others_posts ) 
            ) {
```